drgn/helpers/linux/slab.py

```diff
@@ -53,3 +55,125 @@ def print_slab_caches(prog: Program) -> None:
     for s in for_each_slab_cache(prog):
         name = escape_ascii_string(s.name.string_(), escape_backslash=True)
         print(f"{name} ({s.type_.type_name()})0x{s.value_():x}")
+
+
+def _for_each_slab_page(prog: Program) -> Iterator[Object]:
+    """
+    Iterate over all slab pages
+    :return: Iterator of ``struct page *`` objects.
+    """
+    PGSlab = 1 << prog.constant("PG_slab")
+    for p in for_each_page(prog):
+        try:
+            if p.flags.value_() & PGSlab:
+                yield p
+        except FaultError:
+            pass
+
+
+def get_allocated_slab_objects(slab_cache: Object, obj_type: Type) -> List[Object]:
```

Comment:
Another thing we discussed was making this return pointer objects for consistency with the other `for_each` helpers. I.e, if the caller passes in `struct dentry`, instead of `Object(prog, 'struct dentry', address=0xffff888c8417de00)`, return `Object(prog, 'struct dentry *', value=0xffff888c8417de00)`.

While we're at it, again for consistency with, .e.g., `list_for_each_entry`, let's allow the type to be either a `str` or `Type`. You could do that like:

```python
pointer_type = prog.pointer_type(prog.type(obj_type))
```

Then later when you construct the objects, do:

```python3
Object(prog, pointer_type, addr)
```